### PR TITLE
Free Trials: Add expired state to `PlanFeatures`

### LINKED ISSUE #1471: Free Trials: Add Free Trial Hub "Expired" State
When a free trial expires we are giving the user a 3 day grace period (up to day 17). During that time we need to adjust the Free Trial Hub to indicate that the trial plan has expired and that their features will be removed.

Here is the desktop layout: 

image

For reference here's what the active trial state looks like for the Free Trial Hub: https://github.com/Automattic/wp-calypso/issues/1151#issuecomment-162624443

- [x] Add expired state to `PlanStatus`
- [x] Add expired state to `PlanFeatures` #1905

cc @danhauk @breezyskies @scruffian @Tug

### PR DESCRIPTION
**Pull request by drewblaisdell**
Fixes #1471.

screen shot 2015-12-22 at 3 53 42 pm

I added a couple new props to `PlanFeature` here: `willBeRemoved` and `removalMessage`. I avoided `isInGracePeriod` even though it is currently the same as `willBeRemoved`, the rationale being that it doesn't make sense to describe a _feature_ as part of a grace period, but these prop names still sound slightly awkward to me. If one of you comes up with something better, please mention it.

**Testing**
- Visit http://calypso.localhost:3000/plans/:site for a site with a premium or business trial that has an `expiry` that is one, two, or three days in the future.
- Assert that the version of `PlanFeatures` you see resembles the above. If you have the premium trial, the unlimited themes, Google Analytics, and eCommerce features will not be present.

**Reviews**
- [x] Product
- [x] Code

### CODE REVIEW

**Comment by gziolo** [comment]
Code looks good :+1:

**Comment by drewblaisdell** [comment]
Sorry, I just realized I didn't match the design regarding the progress bar.

**Comment by drewblaisdell** [comment]
Okay, I added a couple commits to render `PlanProgress` in `PlanFeatures`. I updated the screenshot, and the testing instructions are the same.

File: client/lib/plans/index.js [hunk 1]
```diff
@@ -28,3 +28,7 @@ export function getDaysUntilExpiry( plan ) {
 export function isInGracePeriod( plan ) {
 	return getDaysUntilUserFacingExpiry( plan ) <= 0;
 };
+
+export function isInGracePeriod( plan ) {
```
Comment (gziolo) [inline client/lib/plans/index.js]: Is it merge issue? `isInGracePeriod` is here twice.

**Comment by gziolo** [comment]
Still looking good. There is one merge issue to resolve. 
Nice feature :+1:

File: client/lib/plans/index.js (hunk 1 above)
Comment (drewblaisdell) [inline client/lib/plans/index.js]: Yeah, that function was added in two PRs at the same time and was duplicated during the rebase. Nice catch – I removed it.

**Comment by drewblaisdell** [comment]
Thanks @gziolo.

**Comment by spncrb** [comment]
:+1:

**Pull request merged**